wrappers/kn/src/nativeMain/kotlin/zxingcpp/Barcode.kt

```diff
@@ -0,0 +1,137 @@
+package zxingcpp
+
+import cnames.structs.ZXing_Barcode
+import cnames.structs.ZXing_Barcodes
+import kotlinx.cinterop.*
+import zxingcpp.cinterop.*
+import zxingcpp.cinterop.ZXing_ContentType.*
+import kotlin.experimental.ExperimentalNativeApi
+import kotlin.native.ref.createCleaner
+
+@OptIn(ExperimentalForeignApi::class)
+enum class ContentType(internal val cValue: ZXing_ContentType) {
+	Text(ZXing_ContentType_Text),
+	Binary(ZXing_ContentType_Binary),
+	Mixed(ZXing_ContentType_Mixed),
+	GS1(ZXing_ContentType_GS1),
+	ISO15434(ZXing_ContentType_ISO15434),
+	UnknownECI(ZXing_ContentType_UnknownECI);
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun ZXing_ContentType.toKObject(): ContentType {
+	return ContentType.entries.first { it.cValue == this }
+}
+
+data class PointI(
+	val x: Int,
+	val y: Int
+)
+
+@OptIn(ExperimentalForeignApi::class)
+fun ZXing_PointI.toKObject(): PointI = PointI(x, y)
+
+data class Position(
+	val topLeft: PointI,
+	val topRight: PointI,
+	val bottomRight: PointI,
+	val bottomLeft: PointI,
+)
+
+@OptIn(ExperimentalForeignApi::class)
+fun ZXing_Position.toKObject(): Position = Position(
+	topLeft.toKObject(),
+	topRight.toKObject(),
+	bottomRight.toKObject(),
+	bottomLeft.toKObject(),
+)
+
+@OptIn(ExperimentalForeignApi::class)
+class Barcode(val cValue: CValuesRef<ZXing_Barcode>) {
+	val isValid: Boolean
+		get() = ZXing_Barcode_isValid(cValue)
+	val errorMsg: String? by lazy {
```

Comment:
I assume that `isValid` is implicitly read-only because there is only a 'getter' but I wonder what about the `by lazy` properties? Can you assign them a value from outside? That would be inconsistent and undesireable.

Reply:
Is that maybe taken care of by `val` (vs. `var`)?

Reply:
Yes, whether the properties can be modified is controlled by using `val` and `var`.
`val` means value which not modifiable.
`var` means variable.